src/ext/ddtrace.c

```diff
@@ -1254,11 +1257,13 @@ static PHP_FUNCTION(dd_trace_send_traces_via_thread) {
 
     if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "las", &num_traces, &curl_headers,
                                  &payload, &payload_len) == FAILURE) {
-        ddtrace_log_debug("dd_trace_send_traces_via_thread() expects url, http headers, and http body");
+        ddtrace_log_debug("dd_trace_send_traces_via_thread() expects trace count, http headers, and http body");
         RETURN_FALSE
     }
 
-    RETURN_BOOL(ddtrace_send_traces_via_thread(num_traces, curl_headers, payload, payload_len TSRMLS_CC));
+    bool result = ddtrace_send_traces_via_thread(num_traces, curl_headers, payload, payload_len TSRMLS_CC);
+    _ddtrace_prepare_for_new_trace(TSRMLS_C);
```

Comment:
To match my other suggestions:
```suggestion
    dd_prepare_for_new_trace(TSRMLS_C);
```